include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3344,6 +3344,7 @@ struct OrtCustomOp {
   ONNXTensorElementDataType(ORT_API_CALL* GetInputType)(_In_ const struct OrtCustomOp* op, _In_ size_t index);
   size_t(ORT_API_CALL* GetInputTypeCount)(_In_ const struct OrtCustomOp* op);
   ONNXTensorElementDataType(ORT_API_CALL* GetOutputType)(_In_ const struct OrtCustomOp* op, _In_ size_t index);
+  OrtMemType(ORT_API_CALL* GetInputMemoryType)(_In_ const struct OrtCustomOp* op, _In_ size_t index);
   size_t(ORT_API_CALL* GetOutputTypeCount)(_In_ const struct OrtCustomOp* op);
```

Comment:
Is it valid to add things in the middle of this struct, or do any change need to be at the end to not break existing binaries?

e.g. if I update ORT without re-building the calling code the two will have different ideas of what is at this offset in the struct.

Reply:
Yes, please add new members at the end of the struct. E.g. https://github.com/microsoft/onnxruntime/pull/10879.

Reply:
Thanks for pointing out! It is moved to the end of the struct now.